File: tl/src/main/java/com/github/badoualy/telegram/tl/api/storage/TLFilePdf.java
package com.github.badoualy.telegram.tl.api.storage;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLFilePdf extends TLAbsFileType {
    public static final int CONSTRUCTOR_ID = 0xae1e508d;

    public TLFilePdf() {
    }

    @Override
    public String toString() {
        return "storage.filePdf#ae1e508d";
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }

    @Override
    @SuppressWarnings("PointlessBooleanExpression")
    public boolean equals(Object object) {
        if (!(object instanceof TLFilePdf)) return false;
        if (object == this) return true;

        TLFilePdf o = (TLFilePdf) object;

        return true;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/storage/TLFilePdf.kt
package com.github.badoualy.telegram.tl.api.storage

/**
 * storage.filePdf#ae1e508d
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLFilePdf : TLAbsFileType() {
    private val _constructor: String = "storage.filePdf#ae1e508d"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLFilePdf) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0xae1e508d.toInt()
    }
}
